src/routes/blog/post/user-role-guests-missing-scope-account/+page.markdoc

```diff
@@ -0,0 +1,118 @@
+
+# Understanding the error message
+
+Before diving into specific scenarios, let's understand what the error message is telling us:
+
+- **User (role: guests)**: This indicates that the request is coming from an unauthenticated user (a guest) rather than a logged-in user.
+- **missing scope (account)**: This means the operation you're trying to perform requires access to an account.
+
+In essence, this error occurs when you're trying to perform an operation that requires authentication, but Appwrite doesn't detect a valid authenticated session.
+
+There are a handful of reasons you may be receiving this error, which we'll explore in detail below.
+
+# No active session
+
+The most common cause of this error is attempting to access user account information when there's no active session.
+
+If you're not logged in yet and you try to get information about the user with something like `account.get()`, you'll see this error.
+
+```jsx
+// This will fail if no user is logged in
+try {
+  const user = await account.get();
+  console.log(user);
+} catch (error) {
+  console.error(error); // Error: User (role: guests) missing scope (account)
+}
+```
+
+This error is actually expected and useful in certain scenarios, such as when your app first starts and you need to check if a user is logged in. You can catch this error to determine that there's no active session and redirect users to a login page.
+
+However, if you're getting this error when you actually need an authenticated session, simply create a session before calling `account.get()`:
+
+```jsx
+// First create a session
+await account.createEmailSession('user@example.com', 'password');
+
+// Now this will work
+const user = await account.get();
+console.log(user);
+```
+
+# OAuth2 sessions in web applications
+
+A more complex scenario occurs when using OAuth2 authentication in web applications. When you use `account.createOAuth2Session()` in a web app, a cookie is set for the session.
+
+If your app and Appwrite endpoint are on different domains (for example, your app is on `localhost:3000` and you're using `https://cloud.appwrite.io/v1` as your Appwrite endpoint), the cookie may not be included in the `account.get()` request because the cookie is considered a 3rd party cookie.
+
+There are two approaches to solve this issue:
+
+## Option 1: Browser settings and custom domains
+
+For local development:
+
+- Allow 3rd party cookies in your browser settings
+- Some browsers have specific development settings to allow cross-domain cookies during local development
+
+For production:
+
+- [Configure a custom domain](https://appwrite.io/docs/advanced/platform/custom-domains) for your Appwrite project so that cookies are sent from the same domain as your application
+- This ensures that session cookies are properly sent with each request
```

Comment:
These bullet points makes it seem like there are 2 different options but that's not really the case 🧐

Reply:
Updated